msgpack/src/object.cpp

```diff
@@ -0,0 +1,81 @@
+#include <object.hpp>
+#include <adaptor.hpp>
+
+CObject get_cobject(const Value &ark_object, ValueType type)
+{
+	CObject object;
+
+	if(type == ValueType::NFT)
+	{
+		if(ark_object == Ark::True)
+			object = true;
+		else if(ark_object == Ark::False)
+			object = false;
+	}
+	else if(type == ValueType::Number)
+	{
+		object = static_cast<Value>(ark_object).number();
+	}
+	else if(type == ValueType::String)
+	{
+		object = static_cast<Value>(ark_object).string_ref();
```

Comment:
the static_cast<Value>(ark_object) is not needed
and you will have problems with the string since it's not a std::string but a Ark::String (either use Ark::String internally or convert it to std::string with .toString())

Reply:
Up

Reply:
For string and list cast is necessary to remove const qualifier

Reply:
Then use `const_cast` to remove a const qualifer instead

Reply:
> and you will have problems with the string since it's not a std::string but a Ark::String (either use Ark::String internally or convert it to std::string with .toString())